src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,312 @@
+          code: ErrorCode.ProcedureValidationError,
+          message: `There is no Security Token with symbol ${params.symbol}`,
+        })
+      );
+    });
+
+    test('should throw if there is no valid dividend type being provided', async () => {
+      // Instantiate PushDividendPayment with incorrect dividend type
+      target = new PushDividendPayment(
+        { ...params, dividendType: 'wrong' as DividendType },
+        contextMock.getMockInstance()
+      );
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Erc20 dividend module is not attached', async () => {
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      await expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Eth dividend module is not attached', async () => {
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Eth },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      await expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should add a transaction to push a dividend payment of an attached ERC20 dividends module', async () => {
+      // Instantiate PushDividendPayment
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock(
+        'getDividend',
+        Promise.resolve({
+          shareholders: [
+            { address: '0x01', paymentReceived: false },
+            { address: '0x02', paymentReceived: false },
+            { address: '0x03', paymentReceived: true },
+          ],
+        })
+      );
+
+      wrappersMock.mock(
+        'getAttachedModules',
+        Promise.resolve([erc20DividendsMock.getMockInstance()])
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(erc20DividendsMock.getMockInstance().pushDividendPaymentToAddresses)
```

Comment:
This is now only partially correct. You are missing 
`erc20DividendsMock.mock('pushDividendPaymentToAddresses', Promise.resolve('PushDividendPaymentToAddresses'));` -> added somewhere before target.prepareTransactions

Where the resolve has a unique string. This is as per what I mentioned in slack today, let me know if you have more questions.

To test this, as the file is right now, you could change erc20DividendsMock on this line to ethDividendsMock, and it would still pass. That is because without the mock. the functions are all `undefined` type. If you add the code I mention above, you will see that you are no longer able to pass anything else into the calledWith, *only* this.